sql/src/main/java/io/crate/execution/engine/collect/collectors/LuceneOrderedDocCollector.java

```diff
@@ -72,6 +79,11 @@
     private int batchSize;
     private boolean batchSizeReduced = false;
 
+    @Nullable
+    private volatile KillableCollector currentCollector;
+
+    @Nullable
+    private volatile Throwable killed;
```

Comment:
I am not sure if we need both of these to be volatile.

We could probably remove the check here https://github.com/crate/crate/pull/9650/files#diff-1360f05e99db1b25d5f53dec74c7abb2R137 and always depend on the kill raising via the Collector. I think the operations happening between those actions are cheap enough to ignore.

Reply:
> I am not sure if we need both of these to be volatile.

Both can be set and used from different threads.

```
    t1                       t2
                          OrderedDocCollector
                          -> collect()
                            -> set currentCollector
                            -> check killed and raise

  kill()
      ------------------> OrderedDocCollector
                        -> set killed
                        -> check currentCollector and call kill()  if set
```

Reply:
> We could probably remove the check here https://github.com/crate/crate/pull/9650/files#diff-1360f05e99db1b25d5f53dec74c7abb2R137 and always depend on the kill raising via the Collector. I think the operations happening between those actions are cheap enough to ignore.

Yep agree.